pkg/KubeArmorOperator/Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM docker.io/golang:1.20 as builder
+ARG GOARCH
+ARG GOOS
+WORKDIR /app
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# Copy the go source
+
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+COPY api api
+COPY client client
+COPY main.go main.go
+COPY cmd cmd
+COPY common common
+COPY controllers controllers
+COPY enforcer enforcer
+COPY k8s k8s
+COPY runtime runtime
+# Build
+RUN CGO_ENABLED=0 GOOS=${GOOS} GOARCH=${GOARCH} GO111MODULE=on go build -a -o kubearmor-operator main.go
+RUN ln -s kubearmor-operator snitch
```

Comment:
@rksharma95 this symlink would exist only in this build stage. In the final image we'll still have two copies of this binary which will be unnecessary addition to the image size.

Reply:
so to create a symlink in the final image should i use alpine image instead?

Reply:
Why do we need to depend on binary name? Can we just use a flag to differentiate between operator and snitch mode? :thinking: